src/InterventionBackend.php

```diff
@@ -331,11 +267,15 @@ public function loadFrom($path)
     }
 
     /**
+     * @inheritDoc
+     *
      * @param InterventionImage $image
-     * @return $this
      */
-    public function setImageResource($image)
+    public function setImageResource($image): static
     {
+        if (!is_a($image, InterventionImage::class)) {
+            throw new InvalidArgumentException('$image must be an instance of ' . InterventionImage::class);
+        }
```

Comment:
Not technically necessary but since we can't strict type this param it's the next best thing.